### PR TITLE
Update contributing and README with new support scheme

### LINKED ISSUE #3767: Update CONTRIBUTING.md and other instructions with new usage of Discussions vs Issues
[TBC]

### PR DESCRIPTION
**Pull request by astrojuanlu**
## Description

Close #3767.

## Development notes

Simplified the "new issue" form, direct people to Slack or GH issues for support, Discussions for new ideas.

Also removed some unused templates.

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/doc-improvement.md [hunk 1]
```diff
@@ -6,10 +6,10 @@ labels: 'Component: Documentation'
 ---
 
 ## Description
-<!-- Short description of the improvement suggestion. -->
+<!-- Short description of the improvement suggestion -->
 
 ### Documentation page (if applicable)
-<!-- https://docs.kedro.org/en/stable/... -->
+<!-- https://docs.kedro.org/en/0.19.4/... -->
```
[the remark this answers is not in the document]
Comment (astrojuanlu) [reply .github/ISSUE_TEMPLATE/doc-improvement.md:12]: Ugh yeah I think this is a search & replace for something else going wrong. Undoing

**Review by merelcht — COMMENTED** [review]
Generally looks good to me, but I actually use the feature request template quite often myself to create new issues. The bug report template has a lot of fields that are not needed for a non bug issue. How should the team create consistent feature requests going forward if the template is gone?

I think for the sake of making grooming more efficient it's really useful to have a template ready and encourage the team to fill it in properly.

**Comment by astrojuanlu** [comment]
This fell through the cracks a bit but I'll get to it soon 🙏🏼

**Comment by astrojuanlu** [comment]
Will take this again after I'm back from PTO.

**Pull request closed without merging**